pkg/operator/controller/extension/controllerregistration/controllerregistration.go

```diff
@@ -95,47 +88,46 @@ func (r *registration) createOrUpdateControllerRegistration(ctx context.Context,
 						},
 					},
 				},
-			}
-			return nil
-		})
-	return err
-}
-
-// Delete deletes the ControllerRegistration and ControllerDeployment for the given extension.
-func (r *registration) Delete(ctx context.Context, log logr.Logger, virtualClusterClient client.Client, extension *operatorv1alpha1.Extension) error {
-	var (
-		controllerDeployment   = emptyControllerDeployment(extension)
-		controllerRegistration = emptyControllerRegistration(extension)
+			},
+		}
 	)
 
-	log.Info("Deleting ControllerRegistration and ControllerDeployment")
-	if err := kubernetesutils.DeleteObjects(ctx, virtualClusterClient, controllerDeployment, controllerRegistration); err != nil {
+	if v, ok := extension.Annotations[v1beta1constants.AnnotationPodSecurityEnforce]; ok {
+		metav1.SetMetaDataAnnotation(&controllerRegistration.ObjectMeta, v1beta1constants.AnnotationPodSecurityEnforce, v)
+	} else {
+		delete(controllerRegistration.Annotations, v1beta1constants.AnnotationPodSecurityEnforce)
+	}
+
+	data, err := registry.AddAllAndSerialize(controllerDeployment, controllerRegistration)
+	if err != nil {
 		return err
 	}
 
-	log.Info("Waiting until ControllerRegistration is gone")
-	return kubernetesutils.WaitUntilResourceDeleted(ctx, virtualClusterClient, controllerRegistration, 5*time.Second)
+	return managedresources.CreateForShoot(ctx, r.runtimeClient, r.gardenNamespace, managedResourceName(extension), managedresources.LabelValueOperator, false, data)
 }
 
-func emptyControllerRegistration(extension *operatorv1alpha1.Extension) *gardencorev1beta1.ControllerRegistration {
-	return &gardencorev1beta1.ControllerRegistration{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: extension.Name,
-		},
+// Delete deletes the ControllerRegistration and ControllerDeployment for the given extension.
+func (r *registration) Delete(ctx context.Context, log logr.Logger, extension *operatorv1alpha1.Extension) error {
+	mrName := managedResourceName(extension)
+
+	log.Info("Deleting extension registration ManagedResource", "managedResource", client.ObjectKey{Name: mrName, Namespace: r.gardenNamespace})
+	if err := managedresources.DeleteForShoot(ctx, r.runtimeClient, r.gardenNamespace, mrName); err != nil {
+		return fmt.Errorf("failed deleting ManagedResource: %w", err)
```

Comment:
Should we add the managed resource name to the error message?

Reply:
The `Delete` function for managed resources seems to add the names of resources to the error already ([ref](https://github.com/gardener/gardener/blob/09a3faa714119318379f20862f6b8ea2c795b59c/pkg/utils/managedresources/managedresources.go#L234-L278)).